### ISSUE TITLE
Add an option to write mesh related information in a separate .h5 file

### ISSUE THREAD

**Issue by bishtgautam** [RDycore/RDycore#248]
As the mesh becomes larger, it is desirable to output the mesh-related quantities in
a single `.h5` file and not in every `.h5`.

**Comment by knepley** [comment]
It is true that keeping everything together is preferable. We can write the mesh separately, the same way we write it now. The complication is that we would have to change the Load() code to be able to look in another file for DM stuff. Is that worth the complication?

**Comment by bishtgautam** [comment]
Regarding @jedbrown comment:
We can also output multiple timeslices in each file, but the mesh info is becoming too large. For a mesh with 85M cells, if we write out only a single timeslice, the file is 8.5 GB. The 75% of the data in the file is about mesh. I agree that if we always write mesh info into a separate file, there is a chance that we can mistakenly delete the mesh. So, let's add an option to output mesh in a separate file, while the default option would be to add mesh-info in all `.h5` files. Does that seem like a reasonable compromise?

Regarding @knepley's comment:
The mesh we are currently writing out as `.h5` for XDMF isn't what we can read via DM. So, this change isn't going to impact the reading of mesh via DM.

**Comment by jedbrown** [comment]
To clarify, that single-time-step file is 2 GB of solution and 6 GB of mesh (geometry and connectivity). If you use a batch size of 20, then you'll have 46 GB files in which only 6 GB is mesh. This is how we use the CGNS viewer for fluids (albeit those problems are 200x bigger).

**Comment by bishtgautam** [comment]
I agree that adding more time slices will reduce the fraction of mesh info relative to the solution. However, the 46 GB file can only be visualized/analyzed on clusters, and it is not feasible to download such a big file locally for visualization/analysis.

**Comment by knepley** [comment]
> Regarding @knepley's comment: The mesh we are currently writing out as `.h5` for XDMF isn't what we can read via DM. So, this change isn't going to impact the reading of mesh via DM.

Oh, then it does not matter at all and we can do it easily.

**Comment by jeff-cohere** [comment]
I'm doing some I/O work to add diagnostic quantities to output (as part of my mammoth operator refactor), and this shouldn't be too difficult. I'll add it in a small PR after we get through the big refactor PR.
